exporter/elasticsearchexporter/elasticsearch_bulk.go

```diff
@@ -264,48 +251,47 @@ func (p *bulkIndexerPool) Close(ctx context.Context) error {
 }
 
 type worker struct {
-	indexer       *docappender.BulkIndexer
-	items         <-chan esBulkIndexerItem
-	flushInterval time.Duration
-	flushTimeout  time.Duration
-	flushBytes    int
+	indexer      *docappender.BulkIndexer
+	closeCh      <-chan struct{}
+	flushTimeout time.Duration
+	mu           sync.Mutex
+
+	retryBackoff func(int) time.Duration
 
 	stats *bulkIndexerStats
 
 	logger *zap.Logger
 }
 
-func (w *worker) run() {
-	flushTick := time.NewTicker(w.flushInterval)
-	defer flushTick.Stop()
-	for {
+func (w *worker) addBatchAndFlush(ctx context.Context, batch []esBulkIndexerItem) error {
+	w.mu.Lock()
+	defer w.mu.Unlock()
+	for _, item := range batch {
+		if err := w.indexer.Add(item); err != nil {
+			w.logger.Error("error adding item to bulk indexer", zap.Error(err))
+		}
+	}
+	for attempts := 0; ; attempts++ {
+		if err := w.flush(ctx); err != nil {
+			return err
+		} else if w.indexer.Items() == 0 {
+			return nil
+		}
+		backoff := w.retryBackoff(attempts + 1) // TODO: use exporterhelper retry_sender
```

Comment:
Handling retry and timeout in bulk indexer instead of incorporating retry_sender and timeout_sender in this PR, since retry_sender changes may not be straightforward due to leftover document-level 429 in bulk indexer buffer after flush.